### DISCUSSION TITLE
Discussion on persisted useActionData, custom errors, and state mutation

### DISCUSSION THREAD

**Discussion by Centerworx** [Code-Forge-Net/remix-hook-form#16] [General]
I wanted to open a discussion thread to talk about the persisted useActionData, custom errors, and also the reason around mutating the state useForm errors?

**Comment by AlemTuzlak** [comment]
Let me briefly explain what is going on:
When you submit a form via `useRemixForm` it calls `useSubmit` under the hood and submits the data to an action, if that action returns form errors, they are automatically injected into the frontend errors and shown to the user (eg. you returned an email is taken, the validation on the server didn't pass etc). 

The errors are persisted as long as the form is not submitted again (which should not be the case, it should be cleared "artifically" by the package by ignoring the useActionData until the next submit when the user uses `clear()` )
